SMPyBandits/Policies/SWA.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+""" The Sliding Window Average policy for rotting bandits.
+Reference: [Levine , Crammer & Mannor, 2017].
+"""
+from __future__ import division, print_function  # Python 2 compatibility
+
+__author__ = "Julien Seznec"
+__version__ = "0.1"
+
+from math import sqrt, log
+import numpy as np
+import pandas as pd
+np.seterr(divide='ignore')  # XXX dangerous in general, controlled here!
+
+try:
+    from .IndexPolicy import IndexPolicy
+except ImportError:
+    from IndexPolicy import IndexPolicy
+
+
+class SWA(IndexPolicy):
+    """ The Sliding Window Average policy for rotting bandits.
+    Reference: [Levine , Crammer & Mannor, 2017].
+    """
+    def __init__(self, nbArms, horizon=1, subgaussian=1, maxDecrement=1,alpha=0.2, doublingTrick=False):
+        super(SWA, self).__init__(nbArms)
+        self.t = 0
+        self.nbArms = nbArms
+        self.arms_history = {}
+        for arm in range(nbArms):
+            self.arms_history[arm] = np.array([])
```

Comment:
Same, use Python lists instead

Reply:
I am not sure. At each getReward, I need to 1) append the list , 2) sum the current reward to all the reward before (in order to not recompute the sum at each round). 
(1) is more efficient with Python list but (2) is probably more efficient with numpy...

Reply:
In your code below to compute the average, you use `return self.arms_history[arm][int(self.h)] ` which reads ONE value of the arm history, and does not compute an average, so I don't understand this policy

Reply:
This policy selects the arm who gives the largest average of h sample (if one arm do not have h sample, then we select it), . It is equivalent to select the arm which gives the largest sum of h sample (because all the arm have the same h). Yet , it is easier to store and update sums than averages (both computationally and in terms of code). 
A simple implementation would recompute the sum at each round from the history of the arm. Complexity O(Kh).
Instead, I suggest to directly store the sum and to compute it at the reception of the reward. Thus, self.arm_history[arm][h] stores the sum of the h last sample and I use it as index. Therefore at each round I only have to update self.arm_history[arm]. Which can be done in O(h).
I do that by (1) pushing a zero in self.arms_history[arm] and (2) sum the reward to self.arms_history[arm].

Reply:
Okaaay, I finally got the meaning of this line
```python
self.arms_history[arm] = np.insert(self.arms_history[arm], 0, 0) + reward
```